plugins/action/action_impl.cpp

```diff
@@ -391,22 +379,16 @@ void ActionImpl::process_extended_sys_state(const mavlink_message_t &message)
 
 void ActionImpl::loiter_before_takeoff_async(const Action::result_callback_t &callback)
 {
-    uint8_t custom_mode = px4::PX4_CUSTOM_MAIN_MODE_AUTO;
-    uint8_t custom_sub_mode = px4::PX4_CUSTOM_SUB_MODE_AUTO_LOITER;
-
     _parent->set_flight_mode_async(
-        custom_sub_mode, custom_mode,
+        DeviceImpl::HOLD,
         std::bind(&ActionImpl::takeoff_async_continued, this, _1,
                   callback));
 }
 
 void ActionImpl::loiter_before_arm_async(const Action::result_callback_t &callback)
 {
-    uint8_t custom_mode = px4::PX4_CUSTOM_MAIN_MODE_AUTO;
-    uint8_t custom_sub_mode = px4::PX4_CUSTOM_SUB_MODE_AUTO_LOITER;
-
     _parent->set_flight_mode_async(
-        custom_sub_mode, custom_mode,
+        DeviceImpl::HOLD,
```

Comment:
We can rename `HOLD` flight mode to `FlightMode::LOITER` to be consistent  with method name; or vice-versa.

Reply:
The public name of the mode in PX4 is "Hold". The names from DroneCore should match what PX4 uses, which tends to also match what QGC uses in selection lists.